index.d.ts

```diff
@@ -14,6 +14,7 @@ declare namespace LightMyRequest {
     dispatchFunc: DispatchFunc,
     options?: string | InjectOptions
   ): Chain
+  function inject (): Chain
```

Comment:
This is wrong and it throws:

> AssertionError [ERR_ASSERTION]: dispatchFunc should be a function

Reply:
@Eomm This is strange, because:

1) This is how I use it in my code and it works:
```
    const response = await app.inject().get('/health')
```

2) This is how code in question looks like:
```
function inject (dispatchFunc, options, callback) {
  if (typeof callback === 'undefined') {
    return new Chain(dispatchFunc, options)
  } else {
    return doInject(dispatchFunc, options, callback)
  }
}

function Chain (dispatch, option) {
  if (typeof option === 'string') {
    this.option = { url: option }
  } else {
    this.option = Object.assign({}, option)
  }

  this.dispatch = dispatch
  this._hasInvoked = false
  this._promise = null

  if (this.option.autoStart !== false) {
    process.nextTick(() => {
      if (!this._hasInvoked) {
        this.end()
      }
    })
  }
}
```

Why would that throw and not just return a Chain?

Reply:
I assume you are trying to `await` the chain without passing any `dispatchFunc` prior to that. That is not the intention of this particular overload.

Reply:
app is a fastify instance, light my request doesn't know it

```
const inject = require('light-my-request')

// chaining methods
inject()
  .get('/') // set the request method to GET, and request URL to '/'
  .headers({ foo: 'bar' }) // set the request headers
  .query({ foo: 'bar' }) // set the query parameters
  .end((err, res) => {
    console.log(res.payload)
  })
```

Reply:
@Eomm That's my point. In your example calling just `inject()` should be perfectly fine, it just returns a builder on which you keep calling `.get()` etc.
`app` is not relevant in this case.

Reply:
I can change my test to use the whole sequence of
```
inject()
  .get('/') // set the request method to GET, and request URL to '/'
  .headers({ foo: 'bar' }) // set the request headers
  .query({ foo: 'bar' }) // set the query parameters
  .end((err, res) => {
    console.log(res.payload)
  })
```
if you prefer that. Currently even that is not supported by typings.

Reply:
> if you prefer that

It is not matter of preferences, that kind of usage it is not supported and it should not permitted in typings

Reply:
Okay, I adjusted test accordingly